### PR TITLE
Add Workflows SDK scan

### PR DESCRIPTION
**Pull request by alban-stourbe-wmx**
## Proposed changes
Consider Workflows during execution with SDK
Fix issue #5408

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by alban-stourbe-wmx** [comment]
@dogancanbakir  @ehsandeep  ;)

**Comment by GeorginaReeder** [comment]
Thanks for your contribution @alban-stourbe-wmx , we appreciate it!

We also have a [Discord server](https://discord.com/invite/projectdiscovery), which you’re more than welcome to join. It's a great place to connect with fellow contributors and stay updated with the latest developments!

**Comment by alban-stourbe-wmx** [comment]
Hello some news? I need it ;)

File: lib/sdk.go [hunk 1]
```diff
@@ -243,7 +243,15 @@ func (e *NucleiEngine) ExecuteCallbackWithCtx(ctx context.Context, callback ...f
...
-	_ = e.engine.ExecuteScanWithOpts(ctx, e.store.Templates(), e.inputProvider, false)
+	finalTemplates := []*templates.Template{}
+	finalTemplates = append(finalTemplates, e.store.Templates()...)
+	finalTemplates = append(finalTemplates, e.store.Workflows()...)
+
+	if len(finalTemplates) == 0 {
+		return ErrNoTemplatesAvailable
+	}
+
+	_ = e.engine.ExecuteScanWithOpts(ctx, finalTemplates, e.inputProvider, false)
```
Comment (dogancanbakir) [inline lib/sdk.go:254]: Optional:
this 
```go
templatesAndWorkflows := append(e.store.Templates(), e.store.Workflows()...)
_ = e.engine.ExecuteScanWithOpts(ctx, templatesAndWorkflows, e.inputProvider, false)
```
or this
```go
_ = e.engine.ExecuteScanWithOpts(ctx, append(e.store.Templates(), e.store.Workflows()...), e.inputProvider, false)
```

would be better, I guess -no need to be more explicit. Also, I suggest making var name more descriptive.

**Review by dogancanbakir — APPROVED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Comment by alban-stourbe-wmx** [comment]
Thanks guys !

Test Builds have failed, i don't know if it is due to my PR or not

**Comment by GH0st3rs** [comment]
> Test Builds have failed, i don't know if it is due to my PR or not

No, it's not about your PR

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**